docs/source/guides/containerization.rst

```diff
@@ -0,0 +1,602 @@
+================
+Containerization
+================
+
+This guide describes the containerization process and how users can
+define a custom Dockerfile templates to customize Bento container. 
```

Comment:
This guides describes how to customize the Dockerfile generated for containerizing a Bento, by extending BentoML's built-in templates. This is an advanced feature for user to customize container environment that are not directly supported in BentoML. See `Bento Docker Options ` doc to learn about basic options for configuring the Bento generated Docker image.